lib/aws4signer.js

```diff
@@ -73,7 +73,7 @@ const aws4signer = async (esRequest, parent) => {
     esRequest.headers = Object.assign({ host: urlObj.hostname, 'Content-Type': 'application/json' }, esRequest.headers)
     esRequest.path = `${urlObj.pathname}?${urlObj.searchParams.toString()}`
 
-    if (parent.options.openSearchServerless) {
+    if (parent.ESDistribution === "opensearch-serverless") {
       esRequest.headers = Object.assign({ 'X-Amz-Content-SHA256': calculateSHA256(esRequest.body) }, esRequest.headers)
```

Comment:
@ferronrsmith this is wrong. `parent != ElasticSearch` so this condition is always false.

`aws4signer` does not have reference to `ElasticSearch` instance. However blaming this file points me to this commit and line change: https://github.com/elasticsearch-dump/elasticsearch-dump/commit/58023067f10d3e19b16ff0555b5d8b0c5fa00a23#diff-b69fdc248064ec006e5dd5723bb7d584361286f8fc53afdc8e918e8ca25f662aL76

The commit changed from always adding checksum to do so for aoss request. Why is that? Problem here is I don't know how to feed distribution information into here and would be easier if this code path always include body hash.

Reply:
not sure what you're saying. this was all working before. when openSearchServerless is set to true the flag condition kicks in. this is already tested.

Reply:
Let's map this to an enum instead of passing strings

```json
{
  "OPENSEARCH": 'opensearch',
  "ELASTICSEARCH": 'elasticsearch',
  "OPENSEARCH_SERVERLESS": 'opensearch-serverless',
  "ELASTICSEARCH_SERVERLESS": 'opensearch-serverless'
}
```

Reply:
I don't mind changing to enum but what should I do on the hash header?